src/bregman.jl

```diff
@@ -0,0 +1,109 @@
+# Bregman divergence 
+
+"""
+    Implements the Bregman divergence, a friendly introduction to which can be found [here](http://mark.reid.name/blog/meet-the-bregman-divergences.html). Bregman divergences are a minimal generalization of the "mean-minimizer" property. 
+    
+    It is assumed that the function F maps vectors (of any type or size) to real numbers. The inner product used is `Base.dot`, but one can be passed in either by defining `inner` or by passing in a keyword argument. If an analytic gradient isn't available, using a package like `ForwardDiff` is possible. 
+
+    function evaluate(dist::Bregman{T <: Real}, F::Function, p::AbstractVector, q::AbstractVector, ∇::Function; inner=Base.dot)
+"""
+struct Bregman <: PreMetric
+end
+
+function evaluate(dist::Bregman, F::Function, p::AbstractVector, q::AbstractVector, ∇::Function; inner=Base.dot)
+    # Check inputs. 
+    @assert isa(F(p), Real) &&isa(F(q), Real) # Function codomain
+    if size(∇(p)) == size(∇(q)) == size(p) == size(q) # Sizes of vectors 
+    else
+        throw(DimensionMismatch("Either the arrays don't vectors, or the gradients don't match the vectors."))
```

Comment:
This error message is a bit confusing

Reply:
That's because it doesn't make sense :) Fixed.